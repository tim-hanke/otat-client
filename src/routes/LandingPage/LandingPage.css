.LandingPage {
  text-align: center;
}

.LandingPage a {
  background: rgb(255, 121, 121);
  color: white;
  padding: 5px 15px;
  border-radius: 15px;
  margin-left: 10px;
  text-decoration: none;
}

.LandingPage__hint {
  opacity: 0.5;
}

.LandingPage__hint span {
  font-family: monospace;
  background-color: lightgrey;
  opacity: 1;
}

.LandingPage__login {
  background: rgb(255, 121, 121);
  color: white;
  border: 0px;
  padding: 5px 15px;
  border-radius: 15px;
  margin-left: 10px;
  text-decoration: none;
}
