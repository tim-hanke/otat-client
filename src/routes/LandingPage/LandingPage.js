import React, { Component } from "react";
import { Link } from "react-router-dom";
import AuthApiService from "../../services/auth-api-service";
import { Section } from "../../components/Utils/Utils";
import "./LandingPage.css";
import TokenService from "../../services/token-service";

export default class LandingPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  };

  state = { error: null };

  // I put in a button that one-click logs into the demo account
  // to reduce the friction of someone wanting to try out Marker Monkey
  handleDemoLogin = async () => {
    this.setState({ error: null });

    try {
      const res = await AuthApiService.postLogin({
        user_name: "demo",
        password: "Pass123!",
      });
      if (res.authToken) {
        TokenService.saveAuthToken(res.authToken);
      } else {
        throw new Error(res.error);
      }
      this.props.onLogin();
      const { location, history } = this.props;
      const destination = (location.state || {}).from || "/articles";
      history.push(destination);
    } catch (err) {
      this.setState({ error: err.message });
    }
  };

  render() {
    const { error } = this.state;
    return (
      <Section className="LandingPage">
        <h2>
          A place to store all the things you just don't have time to read now.
        </h2>
        <Link to="/register">Register</Link>
        <p className="LandingPage__hint">
          Hint: you can also try out a demo account by clicking below!
        </p>
        <button className="LandingPage__login" onClick={this.handleDemoLogin}>
          Demo Login
        </button>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
      </Section>
    );
  }
}
