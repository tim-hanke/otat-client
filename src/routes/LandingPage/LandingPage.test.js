import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import LandingPage from "./LandingPage";

describe("LandingPage component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <LandingPage />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <LandingPage />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
