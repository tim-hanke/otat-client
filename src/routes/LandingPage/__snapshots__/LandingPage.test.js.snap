// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders the UI as expected 1`] = `
<section
  className="Section LandingPage"
>
  <h2>
    A place to store all the things you just don't have time to read now.
  </h2>
  <a
    href="/register"
    onClick={[Function]}
  >
    Register
  </a>
  <p
    className="LandingPage__hint"
  >
    Hint: you can also try out a demo account by clicking below!
  </p>
  <button
    className="LandingPage__login"
    onClick={[Function]}
  >
    Demo Login
  </button>
  <div
    role="alert"
  />
</section>
`;
